app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -51,21 +67,38 @@ public View onCreateView(LayoutInflater inflater,
 
         // Retrieve the ranking array, passed as argument on instantiation of the class
         roomID = getArguments().getString("roomID");
+        this.drawings =(Bitmap[]) getArguments().getParcelableArray("drawings");
+        this.playerNames = getArguments().getStringArray("playerNames");
         return inflater.inflate(R.layout.ranking_list_fragment, container, false);
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        rankingRef = Database.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".ranking");
-        finishedRef = Database.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".finished");
+
+        currentUserRef = Database.INSTANCE.getReference("users" + "." + Account.getInstance(getActivity().getApplicationContext()).getUserId());
+        rankingRef = Database.INSTANCE.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".ranking");
+        finishedRef = Database.INSTANCE.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".finished");
+        Typeface typeMuro = Typeface.createFromAsset(getActivity().getAssets(), "fonts/Muro.otf");
+        Button button = getActivity().findViewById(R.id.button);
+        button.setTypeface(typeMuro);
+
         retrieveFinalRanking();
     }
 
     public List<String> getRankedUsernames() {
         return rankedUsernames;
     }
 
+    private int getIndexForUserName(String username) throws Exception {
+        for (int i = 0; i < this.playerNames.length; i++) {
+            if(username == playerNames[i]) {
```

Comment:
You shouldn't compare strings by reference. Use string1.equals(string2) instead.